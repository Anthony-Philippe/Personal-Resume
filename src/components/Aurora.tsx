import { Stars } from "@react-three/drei";
import { Canvas } from "@react-three/fiber";
import { memo } from "react";

const AuroraHeroComponent = () => {
  return (
    <div className="absolute inset-0 z-0">
      <Canvas>
        <Stars radius={50} count={750} factor={4} fade speed={2} />
      </Canvas>
    </div>
  );
};

export const AuroraHero = memo(AuroraHeroComponent);