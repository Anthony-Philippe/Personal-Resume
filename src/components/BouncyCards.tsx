import classNames from "classnames";
import { motion } from "framer-motion";
import { memo } from "react";

import image1 from "@assets/images/CNN_project_Python.jpg";
import image3 from "@assets/images/MountainMinder_project_Kotlin.jpg";
import image4 from "@assets/images/Othello_project_C.jpg";
import image2 from "@assets/images/PersonalResume_project_React.jpg";

const BouncyCardsComponent = () => {
  return (
    <section className="mx-auto max-w-5xl px-4 py-12 text-zinc-700 dark:text-slate-100">
      <div className="mb-4 grid grid-cols-12 gap-4">
        <BounceCard className="col-span-12 md:col-span-4" link="https://github.com/Anthony-Philippe/Neural-Network-Image-Classification">
          <CardTitle>Neural Network</CardTitle>
          <CardContent colorFrom="from-indigo-400" colorTo="to-violet-700" imageSrc={image1} />
        </BounceCard>
        <BounceCard className="col-span-12 md:col-span-8" link="https://github.com/Anthony-Philippe/Personal-Resume">
          <CardTitle>Digital Personal Resume</CardTitle>
          <CardContent colorFrom="from-amber-400" colorTo="to-orange-700" imageSrc={image2} />
        </BounceCard>
      </div>
      <div className="grid grid-cols-12 gap-4">
        <BounceCard className="col-span-12 md:col-span-8" link="https://github.com/Anthony-Philippe/MountainMinder/tree/master">
          <CardTitle>Moutain Minder</CardTitle>
          <CardContent colorFrom="from-green-400" colorTo="to-emerald-700" imageSrc={image3} />
        </BounceCard>
        <BounceCard className="col-span-12 md:col-span-4" link="https://github.com/Anthony-Philippe/Othello-Game">
          <CardTitle>Othello Game</CardTitle>
          <CardContent colorFrom="from-pink-400" colorTo="to-red-700" imageSrc={image4} />
        </BounceCard>
      </div>
    </section>
  );
};

export const BouncyCards = memo(BouncyCardsComponent);

interface BounceCardProps {
  className?: string;
  children: React.ReactNode;
  link?: string;
}

const BounceCard = memo(({ className, children, link }: BounceCardProps) => {
  return (
    <motion.div
      whileHover={{ scale: 0.95, rotate: "-1deg" }}
      className={classNames(
        "group relative min-h-[200px] cursor-pointer overflow-hidden rounded-2xl border-2 border-zinc-700 dark:border-slate-50 p-8",
        className
      )}
    >
      <a href={link} target="_blank" rel="noopener noreferrer">{children}</a>
    </motion.div>
  );
});

interface CardTitleProps {
  children: React.ReactNode;
}

const CardTitle = memo(({ children }: CardTitleProps) => {
  return (
    <h3 className="mx-auto text-center text-3xl font-semibold">{children}</h3>
  );
});

interface CardContentProps {
  colorFrom: string;
  colorTo: string;
  imageSrc: string;
}

const CardContent = memo(({ colorFrom, colorTo, imageSrc }: CardContentProps) => {
  return (
    <div
      className={classNames(
        "absolute bottom-0 left-4 right-4 top-14 translate-y-8 rounded-t-2xl p-1 transition-transform duration-[250ms] group-hover:translate-y-4 group-hover:rotate-[2deg]",
        `bg-gradient-to-br ${colorFrom} ${colorTo}`
      )}
    >
      <img
        src={imageSrc}
        alt="Card content"
        className="w-full h-full object-cover rounded-t-xl"
        loading="lazy"
      />
    </div>
  );
});
