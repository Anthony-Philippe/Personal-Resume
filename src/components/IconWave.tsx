import { motion } from "framer-motion";
import { forwardRef, memo, useMemo } from "react";

type IconLink = {
  icon: React.ReactElement;
  link: string;
};

type IconWaveProps = {
  icons: IconLink[];
};

const IconWaveComponent = forwardRef<HTMLDivElement, IconWaveProps>(({ icons }, ref) => {
  const iconElements = useMemo(
    () =>
      icons.map((iconLink, index) => (
        <motion.a
          key={index}
          href={iconLink.link}
          target="_blank"
          rel="noopener noreferrer"
          initial={{ y: "-100vh", opacity: 0 }}
          animate={{ y: 0, opacity: 1 }}
          transition={{ duration: 1.2, delay: index * 0.2 }}
          className="relative cursor-pointer text-3xl text-white"
          whileHover={{ scale: 1.3, transition: { duration: 0.4 } }}
        >
          {iconLink.icon}
        </motion.a>
      )),
    [icons]
  );

  return (
    <motion.div ref={ref} className="flex space-x-8">
      {iconElements}
    </motion.div>
  );
});

export const IconWave = memo(IconWaveComponent);