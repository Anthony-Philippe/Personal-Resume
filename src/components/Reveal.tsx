import { motion, useAnimation, useInView } from "framer-motion";
import { memo, useCallback, useEffect, useMemo, useRef } from "react";

interface RevealProps {
  children: JSX.Element;
  width?: "fit-content" | "100%";
}

const RevealComponent = ({ children, width = "fit-content" }: RevealProps) => {
  const ref = useRef<HTMLDivElement>(null);
  const isInView = useInView(ref, { once: true });

  const mainControls = useAnimation();
  const slideControls = useAnimation();

  const startAnimation = useCallback(() => {
    if (isInView) {
      mainControls.start("visible");
      slideControls.start("visible");
    }
  }, [isInView, mainControls, slideControls]);

  useEffect(() => {
    startAnimation();
  }, [startAnimation]);

  const mainVariants = useMemo(() => ({
    hidden: { opacity: 0, y: 75 },
    visible: { opacity: 1, y: 0 },
  }), []);

  const slideVariants = useMemo(() => ({
    hidden: { left: 0 },
    visible: { left: "100%" },
  }), []);

  return (
    <div ref={ref} style={{ position: "relative", width, overflow: "hidden" }}>
      <motion.div
        variants={mainVariants}
        initial="hidden"
        animate={mainControls}
        transition={{ duration: 0.5, delay: 0.5 }}
      >
        {children}
      </motion.div>
      <motion.div
        variants={slideVariants}
        initial="hidden"
        animate={slideControls}
        transition={{ duration: 0.8, ease: "easeIn" }}
        className="absolute inset-0 top-1 z-20 bg-primary-500"
      />
    </div>
  );
};

export const Reveal = memo(RevealComponent);