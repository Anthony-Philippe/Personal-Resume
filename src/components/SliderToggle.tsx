import Brightness2OutlinedIcon from "@mui/icons-material/Brightness2Outlined";
import WbSunnyOutlinedIcon from "@mui/icons-material/WbSunnyOutlined";
import { motion } from "framer-motion";
import { useState } from "react";

const TOGGLE_CLASSES =
  "text-sm font-medium flex items-center gap-2 px-3 md:pl-3 md:pr-3.5 py-3 md:py-1.5 transition-colors relative z-10";

export const SliderToggle = () => {
  const [selected, setSelected] = useState("dark");
  return (
    <div
      className={`grid transition-colors ${
        selected === "light" ? console.log("light") : console.log("dark")
      }`}
    >
      <Switch selected={selected} setSelected={setSelected} />
    </div>
  );
};

interface SwitchProps {
  selected: string;
  setSelected: (selected: string) => void;
}

const Switch = ({ selected, setSelected }: SwitchProps) => {
  const isSelectedLight = selected === "light";
  const isSelectedDark = selected === "dark";

  return (
    <div className="relative flex w-fit items-center rounded-full">
      <button
        className={`${TOGGLE_CLASSES} ${
          isSelectedLight ? "text-white" : "text-slate-300"
        }`}
        onClick={() => {
          setSelected("light");
        }}
      >
        <Brightness2OutlinedIcon className="relative z-10 text-lg md:text-sm" />
        <span
          className={`relative z-10 font-semibold ${
            isSelectedLight ? "text-white" : "text-slate-300"
          }`}
        >
          Light
        </span>
      </button>
      <button
        className={`${TOGGLE_CLASSES} ${
          isSelectedDark ? "text-white" : "text-slate-300"
        }`}
        onClick={() => {
          setSelected("dark");
        }}
      >
        <WbSunnyOutlinedIcon className="relative z-10 text-lg md:text-sm" />
        <span
          className={`relative z-10 font-semibold ${
            isSelectedDark ? "text-white" : "text-slate-300"
          }`}
        >
          Dark
        </span>
      </button>
      <div
        className={`absolute inset-0 z-0 flex ${
          isSelectedDark ? "justify-end" : "justify-start"
        }`}
      >
        <motion.span
          layout
          transition={{ type: "spring", damping: 15, stiffness: 250 }}
          className="h-full w-1/2 rounded-full bg-gradient-to-r from-violet-600 to-indigo-600"
        />
      </div>
    </div>
  );
};
