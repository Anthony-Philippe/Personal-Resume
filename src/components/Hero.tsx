import { Reveal } from "./Reveal";

export const Hero = () => {
  return (
    <section className="max-w-screen-sm mx-auto p-4">
      <Reveal>
        <h1 className="text-5xl font-semibold mb-2">Hey, I'm John Doe<span>!</span></h1>
      </Reveal>
      <Reveal>
        <h2 className="text-xl font-light mb-4">
          I'm a <span>Web Developer</span>
        </h2>
      </Reveal>
      <Reveal>
        <p className="text-sm font-extralight">
          I design and develop services for customers of all sizes, specializing
          in creating stylish, modern websites, web services and online stores.
          I design and develop services for customers of all sizes, specializing
          in creating stylish, modern websites, web services and online stores.
        </p>
      </Reveal>
    </section>
  );
};
