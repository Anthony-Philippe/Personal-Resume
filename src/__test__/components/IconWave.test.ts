import { IconWave } from "../../components/IconWave";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  createElement: jest.fn(),
}));

jest.mock("@mui/icons-material/GitHub", () => () => "GitHubIcon");
jest.mock("@mui/icons-material/LinkedIn", () => () => "LinkedInIcon");

describe("IconWave", () => {
  test("should render without icons", () => {
    const wrapper = IconWave({ icons: [] });
    expect(wrapper).toMatchSnapshot();
  });

  test("should render with icons", () => {
    const icons = [
      { icon: "GitHubIcon", link: "https://github.com" },
      { icon: "LinkedInIcon", link: "https://linkedin.com" },
    ];
    const wrapper = IconWave({ icons } as any);
    expect(wrapper).toMatchSnapshot();
  });

  test("should render framer motion", () => {
    jest.mock("framer-motion", () => ({
      ...jest.requireActual("framer-motion"),
      motion: {
        div: ({ children }: { children: any }) => children,
      },
    }));
  });
});
