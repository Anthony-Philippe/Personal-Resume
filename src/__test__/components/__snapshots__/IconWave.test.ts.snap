// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconWave should render with icons 1`] = `
<div
  className="flex space-x-8"
>
  <ForwardRef(MotionComponent)
    animate={
      {
        "opacity": 1,
        "y": 0,
      }
    }
    className="relative text-3xl text-white cursor-pointer"
    href="https://github.com"
    initial={
      {
        "opacity": 0,
        "y": "-100vh",
      }
    }
    rel="noopener noreferrer"
    target="_blank"
    transition={
      {
        "delay": 0,
        "duration": 1.2,
      }
    }
    whileHover={
      {
        "scale": 1.3,
        "transition": {
          "duration": 0.4,
        },
      }
    }
  >
    GitHubIcon
  </ForwardRef(MotionComponent)>
  <ForwardRef(MotionComponent)
    animate={
      {
        "opacity": 1,
        "y": 0,
      }
    }
    className="relative text-3xl text-white cursor-pointer"
    href="https://linkedin.com"
    initial={
      {
        "opacity": 0,
        "y": "-100vh",
      }
    }
    rel="noopener noreferrer"
    target="_blank"
    transition={
      {
        "delay": 0.2,
        "duration": 1.2,
      }
    }
    whileHover={
      {
        "scale": 1.3,
        "transition": {
          "duration": 0.4,
        },
      }
    }
  >
    LinkedInIcon
  </ForwardRef(MotionComponent)>
</div>
`;

exports[`IconWave should render without icons 1`] = `
<div
  className="flex space-x-8"
/>
`;
